--- a/src/widgets/kernel/qtooltip.cpp
+++ b/src/widgets/kernel/qtooltip.cpp
@@ -379,9 +379,13 @@
 /** \internal
   Cleanup the _q_stylesheet_parent property.
  */
+#if QT_CONFIG(style_stylesheet)
 void QTipLabel::styleSheetParentDestroyed()
 {
     setProperty("_q_stylesheet_parent", QVariant());
     styleSheetParent = nullptr;
 }
+#else
+void QTipLabel::styleSheetParentDestroyed() {}
+#endif
 
 /*!
     Shows \a text as a tool tip, with the global position \a pos as
