--- a/buildrun.cxx	2023-05-03 10:17:44.338633166 +0200
+++ b/buildrun.cxx	2023-05-03 10:18:48.465507072 +0200
@@ -943,7 +943,8 @@
           if (remotedir.empty())
             opt_u.append(s.uprobes_path);
           else
-            opt_u.append(remotedir + "/" + basename(s.uprobes_path.c_str()));
+            //opt_u.append(remotedir + "/" + basename(s.uprobes_path.c_str()));
+            opt_u.append(remotedir + "/" + s.uprobes_path.substr(s.uprobes_path.rfind('/')+1));
         }
       cmd.push_back(opt_u);
     }
--- a/remote.cxx	2023-05-03 13:52:06.614257954 +0200
+++ b/remote.cxx	2023-05-03 13:53:19.784994466 +0200
@@ -472,7 +472,8 @@
 
         if (!s->uprobes_path.empty())
           {
-            string remoteuprobes = basename(s->uprobes_path.c_str());
+            //string remoteuprobes = basename(s->uprobes_path.c_str());
+            string remoteuprobes = s->uprobes_path.substr(s->uprobes_path.rfind('/')+1);
             if ((rc = send_file(s->uprobes_path, remoteuprobes)))
               return rc;
 
