--- a/util.cxx	2023-05-03 13:20:34.930972141 +0200
+++ b/util.cxx	2023-05-03 13:21:15.398273958 +0200
@@ -257,8 +257,10 @@
 {
   char *dirc, *basec, *bname, *dname;
 
-  dirc = strdupa (path.c_str());
-  basec = strdupa (path.c_str());
+  dirc = strdup (path.c_str());
+  basec = strdup (path.c_str());
   dname = dirname (dirc);
   bname = basename (basec);
+  free(dirc);
+  free(basec);
   directory = dname;
