--- a/staprun/staprun.h	2023-05-03 21:59:59.097825632 +0200
+++ b/staprun/staprun.h	2023-05-03 22:00:06.457699309 +0200
@@ -30,7 +30,7 @@
 #include <errno.h>
 #include <linux/fd.h>
 #include <sys/mman.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <pthread.h>
 #include <sys/socket.h>
 #include <linux/limits.h>
--- a/stapbpf/stapbpf.cxx	2023-05-03 22:02:20.483398963 +0200
+++ b/stapbpf/stapbpf.cxx	2023-05-03 22:02:34.879151884 +0200
@@ -34,7 +34,7 @@
 #include <limits.h>
 #include <inttypes.h>
 #include <getopt.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/syscall.h>
