--- a/util.h	2023-05-03 08:32:15.445228767 +0200
+++ b/util.h	2023-05-03 08:33:38.683781936 +0200
@@ -16,6 +16,27 @@
 #include <map>
 #include <algorithm>
 #include <limits>
+#include <sys/types.h>
+#include <linux/types.h>
+
+typedef off_t     __off_t;
+typedef signed char __int8_t;
+typedef unsigned char __uint8_t;
+typedef signed short int __int16_t;
+typedef unsigned short int __uint16_t;
+typedef signed int __int32_t;
+typedef unsigned int __uint32_t;
+#if __WORDSIZE == 64
+typedef signed long int __int64_t;
+typedef unsigned long int __uint64_t;
+#elif defined(__GNUC__)
+__extension__ typedef signed long long int __int64_t;
+__extension__ typedef unsigned long long int __uint64_t;
+#endif
+
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
 
 extern "C" {
 #if ENABLE_NLS
@@ -415,6 +415,9 @@
           const sigset_t *sigmask);
 #endif
 
+#ifndef __GLIBC__
+#define canonicalize_file_name(name) realpath(name,NULL)
+#endif
 
 int read_from_file (const std::string &fname, int &data);
 template <class T>
--- a/tapsets.cxx	2023-05-03 09:17:52.009600253 +0200
+++ b/tapsets.cxx	2023-05-03 09:18:02.989408144 +0200
@@ -51,7 +51,6 @@
 #include <elfutils/libdw.h>
 #include <dwarf.h>
 #include <elf.h>
-#include <obstack.h>
 #include <glob.h>
 #include <fnmatch.h>
 #include <stdio.h>
--- a/staprun/staprun.h	2023-05-03 21:56:21.277564278 +0200
+++ b/staprun/staprun.h	2023-05-03 21:57:10.364721740 +0200
@@ -48,6 +48,7 @@
 #include <locale.h>
 #endif
 
+typedef off_t     __off_t;
 
 /* define gettext options if NLS is set */
 #if ENABLE_NLS
