--- a/util/logging.c
+++ b/util/logging.c
@@ -53,11 +53,11 @@ static void print_detail_line(FILE* log_file, const char* log_level,
     int _strlen;
 
     if (func_name != NULL && func_name[0] != 0) {
-	    snprintf(format_buffer, LOG_FORMAT_BUFFER_SIZE, "[%s][%s](%s) %s",
-	            get_formatted_date(), log_level, func_name, log_format);
+	    snprintf(format_buffer, LOG_FORMAT_BUFFER_SIZE, "[%s](%s) %s",
+	            log_level, func_name, log_format);
 	} else {
-	    snprintf(format_buffer, LOG_FORMAT_BUFFER_SIZE, "[%s][%s] %s",
-	             get_formatted_date(), log_level, log_format);
+	    snprintf(format_buffer, LOG_FORMAT_BUFFER_SIZE, "[%s] %s",
+	            log_level, log_format);
 	}
 
 	/* Append a newline if not exist */
