--- a/amule/Makefile
+++ b/amule/Makefile
@@ -31,6 +31,19 @@ define Package/amule
 	+libupnp +libbfd +libcryptopp
 endef
 
+define Download/AmuleWebUI
+  URL:=https://github.com/MatteoRagni/AmuleWebUI-Reloaded/archive
+  URL_FILE:=master.zip
+  FILE:=AmuleWebUI.zip
+  HASH:=skip
+endef
+
+define Build/Prepare
+	$(call Build/Prepare/Default)
+	unzip $(DL_DIR)/AmuleWebUI.zip -d $(PKG_BUILD_DIR)
+	
+endef
+
 CONFIGURE_ARGS+= \
 	--disable-nls \
 	--disable-static \
@@ -72,12 +85,16 @@ endef
 
 define Package/amule/install
 	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) $(1)/usr/share/amule/webserver
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/amulecmd $(1)/usr/bin/amulecmd
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/amuled $(1)/usr/bin/amuled
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/amuleweb $(1)/usr/bin/amuleweb
+	$(CP) $(PKG_BUILD_DIR)/src/webserver/default $(1)/usr/share/amule/webserver
+	$(CP) $(PKG_BUILD_DIR)/AmuleWebUI-Reloaded-master $(1)/usr/share/amule/webserver/AmuleWebUI-Reloaded
 
 	$(INSTALL_DIR) $(1)/usr/share
 	$(CP) $(PKG_INSTALL_DIR)/usr/share/amule $(1)/usr/share/
 endef
 
+$(eval $(call Download,AmuleWebUI))
 $(eval $(call BuildPackage,amule))

--- a/luci-app-amule/root/etc/config/amule
+++ b/luci-app-amule/root/etc/config/amule
@@ -62,7 +62,7 @@ config amule 'main'
         option web_enabled '0'
         option upnp_enabled '0'
         option proxy_type 'socks5'
-        option template 'default'
+        option template 'AmuleWebUI-Reloaded'
         option use_low_rights_user '0'
         option web_port '44711'
         option upnp_web_server_enabled '0'
