--- a/luci-theme-argon/ucode/template/themes/argon/footer.ut
+++ b/luci-theme-argon/ucode/template/themes/argon/footer.ut
@@ -19,13 +19,6 @@
 		</div>
 		<footer class="mobile-hide" style="text-wrap: auto">
 			<div class="footer-content" style="display: flex; flex-wrap: wrap; gap: 0.5em; justify-content: end;">
-				<a class="luci-link" href="https://github.com/openwrt/luci" target="_blank">Powered by {{ version.luciname }} ({{ version.luciversion }})</a>
-
-				<span class="footer-separator">|</span>
-
-				<a href="https://github.com/jerrykuku/luci-theme-argon" target="_blank">ArgonTheme {# vPKG_VERSION #}</a>
-
-				<span class="footer-separator">|</span>
 
 				<a class="luci-link" href="{{ version.disturl }}" target="_blank">{{ version.distname }} {{ version.distversion }}-{{ version.distrevision }}</a>
 
