--- a/automount/Makefile
+++ b/automount/Makefile
@@ -20,6 +20,7 @@ define Package/automount
   TITLE:=Mount autoconfig hotplug script.
   MAINTAINER:=Lean
   DEPENDS:= \
+    +luci-app-partexp \
     +block-mount \
     +e2fsprogs \
     +kmod-usb-storage \
@@ -48,10 +49,6 @@ endef
 
 define Package/automount/install
 	$(INSTALL_DIR) $(1)/etc/uci-defaults
-	$(INSTALL_BIN) ./files/11-anonmount $(1)/etc/uci-defaults/
-
-	$(INSTALL_DIR) $(1)/etc/hotplug.d/block
-	$(INSTALL_BIN) ./files/15-automount $(1)/etc/hotplug.d/block/
 endef
 
 define Package/ntfs3-mount/install
