--- a/luci-app-easytier/Makefile
+++ b/luci-app-easytier/Makefile
@@ -10,7 +10,7 @@ PKG_VERSION:=2.3.0
 PKG_RELEASE:=1
 
 LUCI_TITLE:=LuCI support for EasyTier
-LUCI_DEPENDS:=+kmod-tun +luci-compat
+LUCI_DEPENDS:=+kmod-tun +easytier +luci-compat
 LUCI_PKGARCH:=all
 
 PKG_NAME:=luci-app-easytier

--- a/luci-app-samba4/Makefile
+++ b//luci-app-samba4/Makefile
@@ -3,7 +3,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=Network Shares - Samba 4 SMB/CIFS fileserver
-LUCI_DEPENDS:=+luci-base +samba4-server
+LUCI_DEPENDS:=+luci-base +samba4-server +autoshare-samba
 
 PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Andy Walsh <andy.walsh44+github@gmail.com>

--- a/luci-app-ksmbd/Makefile
+++ b/luci-app-ksmbd/Makefile
@@ -3,7 +3,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=Network Shares - Ksmbd the SMB kernel fileserver
-LUCI_DEPENDS:=+luci-base +ksmbd-server
+LUCI_DEPENDS:=+luci-base +ksmbd-server +autoshare-ksmbd
 
 PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Richard Yu <yurichard3839@gmail.com>

--- a/fullconenat/Makefile
+++ b/fullconenat/Makefile
@@ -36,7 +36,7 @@ define Package/ip6tables-mod-fullconenat
   SECTION:=net
   CATEGORY:=Network
   TITLE:=FULLCONENAT ip6tables extension
-  DEPENDS:=ip6tables +kmod-nf-nat6 +kmod-ipt-fullconenat +ip6tables-mod-nat
+  DEPENDS:=+ip6tables +kmod-nf-nat6 +kmod-ipt-fullconenat +ip6tables-mod-nat
 endef
 
 define KernelPackage/ipt-fullconenat

--- a/phtunnel/Makefile
+++ b/phtunnel/Makefile
@@ -36,7 +36,6 @@ PKG_SOURCE:=phtunnel
 PKG_SOURCE_URL:=https://mirrors.oray.com/orayos/packages/$(PKG_NAME)/$(PKG_ARCH_PHT)/$(PKG_VERSION)/bin
 PKG_HASH:=skip
 
-PKG_FLAGS:=nonshared
 PKG_MAINTAINER:=Oray <developer@oray.com>
 
 include $(INCLUDE_DIR)/package.mk

--- a/pgyvpn/Makefile
+++ b/pgyvpn/Makefile
@@ -36,7 +36,6 @@ PKG_SOURCE:=pgyvpnsvr
 PKG_SOURCE_URL:=https://mirrors.oray.com/orayos/packages/$(PKG_NAME)/$(PKG_ARCH_PGYYPN)/$(PKG_VERSION)/bin
 PKG_HASH:=skip
 
-PKG_FLAGS:=nonshared
 PKG_MAINTAINER:=Oray <developer@oray.com>
 
 include $(INCLUDE_DIR)/package.mk

--- a/luci-app-rclone/Makefile
+++ b/luci-app-rclone/Makefile
@@ -9,7 +9,7 @@ PKG_VERSION:=1.4.1
 PKG_RELEASE:=4
 
 LUCI_TITLE:=LuCI support for Rclone
-LUCI_DEPENDS:=+luci-compat +rclone-config \
+LUCI_DEPENDS:=+luci-compat +rclone-config +fuse-overlayfs +fuse3-utils \
               +PACKAGE_$(PKG_NAME)_INCLUDE_rclone-webui:rclone-webui-react \
               +PACKAGE_$(PKG_NAME)_INCLUDE_rclone-ng:rclone-ng
 

--- a/k3screenctrl/Makefile
+++ b/k3screenctrl/Makefile
@@ -19,7 +19,7 @@ TARGET_CFLAGS+= -D_GNU_SOURCE
 define Package/k3screenctrl
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS:=@TARGET_bcm53xx_generic_DEVICE_phicomm_k3 +@KERNEL_DEVMEM
+  DEPENDS:=@(TARGET_bcm53xx_generic_DEVICE_phicomm_k3||TARGET_DEVICE_bcm53xx_generic_DEVICE_phicomm_k3) +@KERNEL_DEVMEM
   TITLE:=LCD screen controller on PHICOMM K3
   URL:=https://github.com/updateing/k3-screen-ctrl
 endef

--- a/luci-app-packet-capture/Makefile
+++ b/luci-app-packet-capture/Makefile
@@ -4,7 +4,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=Packet capture application
-LUCI_DEPENDS:=+luci-mod-admin-full +tcpdump +uhttpd-mod-ubus +coreutils +coreutils-timeout
+LUCI_DEPENDS:=+luci-mod-admin-full +tcpdump +coreutils +coreutils-timeout
 
 PKG_MAINTAINER:=Wojciech Jowsa <wojciech.jowsa@gmail.com>
 PKG_LICENSE:=Apache-2.0

--- a/luci-app-smartdns/Makefile
+++ b/luci-app-smartdns/Makefile
@@ -18,6 +18,7 @@ PKG_CONFIG_DEPENDS:=CONFIG_PACKAGE_luci-app-smartdns_INCLUDE_WebUI
 define Package/luci-app-smartdns/config
   config PACKAGE_luci-app-smartdns_INCLUDE_WebUI
     bool "Include WebUI support"
+    default y
 endef
 
 include ../../luci.mk
