--- a/luci-app-transmission/Makefile
+++ b/luci-app-transmission/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI Support for Transmission
-LUCI_DEPENDS:=+luci-base +transmission-daemon
+LUCI_DEPENDS:=+luci-base +transmission-daemon +transmission-web-control
 
 PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Richard Yu <yurichard3839@gmail.com>

--- a/luci-app-transmission/htdocs/luci-static/resources/view/transmission.js
+++ b/luci-app-transmission/htdocs/luci-static/resources/view/transmission.js
@@ -34,7 +34,7 @@ return view.extend({
 		var webinstalled = res[1] || !!uci.get_first('transmission', 'transmission', 'web_home');
 
 		var button = '';
-		if (running && webinstalled)
+		if (running)
 			button = '&#160;<a class="btn" href="http://' + window.location.hostname + ':' + port + '" target="_blank" rel="noreferrer noopener">' + _('Open Web Interface') + '</a>';
 
 		var m, s, o;
