--- a/luci-app-store/luasrc/controller/store.lua
+++ b/luci-app-store/luasrc/controller/store.lua
@@ -246,7 +246,7 @@ function store_action(param)
         local metapkg = pkg and (metapkgpre .. pkg) or ""
         if action == "update" or pkg then
             if action == "update" or action == "install" then
-                code, out, err = _action(myopkg, action, metapkg)
+                code, out, err = _action(myopkg, action .. " --force-overwrite --force-checksum --force-depends", metapkg)
             else
                 local meta = json_parse(fs.readfile(metadir .. "/" .. pkg .. ".json"))
                 local pkgs = {}
@@ -313,7 +313,7 @@ function store_upload()
         if string.lower(string.sub(path, -4, -1)) == ".run" then
             code, out, err = _action("sh", "-c", "ls -l \"%s\"; md5sum \"%s\" 2>/dev/null; chmod 755 \"%s\" && \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path, path, path, path })
         else
-            code, out, err = _action("sh", "-c", "opkg install \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path })
+            code, out, err = _action("sh", "-c", "opkg install --force-overwrite --force-checksum --force-depends \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path })
         end
     else
         code = 500
