--- a/luci-base/ucode/dispatcher.uc
+++ b/luci-base/ucode/dispatcher.uc
@@ -939,7 +939,12 @@ dispatch = function(_http, path) {
 				let cookie_name = (http.getenv('HTTPS') == 'on') ? 'sysauth_https' : 'sysauth_http',
 				    cookie_secure = (http.getenv('HTTPS') == 'on') ? '; secure' : '';
 
-				http.header('Set-Cookie', `${cookie_name}=${session.sid}; path=${build_url()}; SameSite=strict; HttpOnly${cookie_secure}`);
+				let cookie_p = uci.get('wizard', 'default', 'cookie_p');
+				if (cookie_p == '0') {
+					http.header('Set-Cookie', `${cookie_name}=${session.sid}; path=${build_url()}; SameSite=strict; HttpOnly${cookie_secure}`);
+				} else {
+					http.header('Set-Cookie', `${cookie_name}=${session.sid}; Max-Age=2147483647; path=${build_url()}; SameSite=strict; HttpOnly${cookie_secure}`);
+				}
 				http.redirect(build_url(...resolved.ctx.request_path));
 
 				return;

--- a/luci-base/root/www/index.html
+++ b/luci-base/root/www/index.html
@@ -15,6 +15,5 @@
 		</style>
 	</head>
 	<body>
-		<a href="cgi-bin/luci/">LuCI - Lua Configuration Interface</a>
 	</body>
 </html>

--- a/luci-base/htdocs/luci-static/resources/ui.js
+++ b/luci-base/htdocs/luci-static/resources/ui.js
@@ -3129,12 +3129,6 @@ var UIMenu = baseclass.singleton(/** @lends LuCI.ui.menu.prototype */ {
 
 				for (var i = 0; root != null && i < path.length; i++)
 					root = L.isObject(root.children) ? root.children[path[i]] : null;
-
-				if (root)
-					subnode = Object.assign({}, subnode, {
-						children: root.children,
-						action: root.action
-					});
 			}
 
 			children.push(subnode);
