--- a/luci-app-nodogsplash/Makefile
+++ b/luci-app-nodogsplash/Makefile
@@ -36,9 +36,6 @@ define Package/luci-app-nodogsplash/install
 	$(INSTALL_DIR) $(1)/www/luci-static/resources/view/nodogsplash/
 	$(INSTALL_DATA) ./htdocs/luci-static/resources/view/nodogsplash/*.js $(1)/www/luci-static/resources/view/nodogsplash/
 
-	$(INSTALL_DIR) $(1)/usr/libexec/rpcd/
-	$(INSTALL_DATA) ./root/usr/libexec/rpcd/luci.nodogsplash $(1)/usr/libexec/rpcd/
-
 	$(INSTALL_DIR) $(1)/usr/share/luci/menu.d/
 	$(INSTALL_DATA) ./root/usr/share/luci/menu.d/luci-app-nodogsplash.json $(1)/usr/share/luci/menu.d/

