--- a/routergo/Makefile
+++ b/routergo/Makefile
@@ -27,7 +27,7 @@ define Package/$(PKG_NAME)
 	CATEGORY:=Network
 	SUBMENU:=Web Servers/Proxies
 	TITLE:=RouterGo
-	DEPENDS:=@(x86_64||aarch64||arm) +docker
+	DEPENDS:=@(x86_64||aarch64||arm) +docker +luci-app-linkease
 	URL:=https://github.com/Carseason/openwrt-packages
 endef
 

--- a/luci-app-routerdog/luasrc/controller/routerdog.lua
+++ b/luci-app-routerdog/luasrc/controller/routerdog.lua
@@ -2,7 +2,7 @@ module("luci.controller.routerdog", package.seeall)
 
 function index()
     if nixio.fs.access("/usr/lib/lua/luci/view/routerdog/main.htm") then
-        entry({"admin", "routerdog"}, template("routerdog/index"), _("RouterDog"), 1).leaf = true
+        entry({"admin", "routerdog"}, template("routerdog/index")).leaf = true
     end
     entry({"admin", "routerdog_api"}, call("redirect_index"))
     entry({"admin", "routerdog_api", "status"}, call("routerdog_api_status"))

--- a/luci-app-routerdog/root/usr/share/luci/menu.d/luci-app-routerdog.json
+++ /dev/null
@@ -1,10 +0,0 @@
-{
-    "admin/routerdog": {
-        "title": "RouterDog",
-        "order": -1,
-        "action": {
-            "type": "template",
-            "path": "routerdog/index"
-        }
-    }
-}
\ No newline at end of file

--- a/luci-app-routerdog/root/etc/uci-defaults/50_luci-routerdog
+++ b/luci-app-routerdog/root/etc/uci-defaults/50_luci-routerdog
@@ -1,4 +1,7 @@
 #!/bin/sh
 
+uci set wizard.default.landing_page='routerdog'
+uci commit wizard
+
 rm -f /tmp/luci-indexcache
 exit 0
