From 96fa20f4e585d93a289b1abb610920cc8d48faf4 Mon Sep 17 00:00:00 2001
From: sbwml <admin@cooluc.com>
Date: Sat, 2 May 2026 10:56:06 +0800
Subject: [PATCH] tproxy: refactor get_pid_pname to use bpf_get_current_comm

Signed-off-by: sbwml <admin@cooluc.com>
---
 dae-core/control/kern/tproxy.c | 70 +++----------------------------------------
 1 file changed, 4 insertions(+), 66 deletions(-)

--- a/dae-core/control/kern/tproxy.c
+++ b/dae-core/control/kern/tproxy.c
@@ -1730,75 +1730,13 @@ int tproxy_dae0_ingress(struct __sk_buff
 	return bpf_redirect(redirect_entry->ifindex, flags);
 }
 
-struct get_real_comm_ctx {
-	char *arg_buf;
-	u8 l;
-};
-
-static int __noinline get_real_comm_loop_cb(__u32 index, void *data)
-{
-	/*
-	* For string like: /usr/lib/sddm/sddm-helper --socket /tmp/sddm-auth1
-	* We extract "sddm-helper" from it.
-	*/
-	struct get_real_comm_ctx *ctx = (struct get_real_comm_ctx *)data;
-
-	if (index >= MAX_ARG_LEN) // always false, just to make verifier happy
-		return 1;
-	if (unlikely(ctx->arg_buf[index] == '/'))
-		ctx->l = index + 1;
-	if (unlikely(ctx->arg_buf[index] == ' ' ||
-		     ctx->arg_buf[index] == '\0')) {
-		// Write to dst.
-		ctx->arg_buf[index] = '\0';
-		return 1;
-	}
-	return 0;
-}
-
 /// Parse command line arguments to get the real command name and tgid.
 static __always_inline int get_pid_pname(struct pid_pname *pid_pname)
 {
-	int ret;
-	// Get pointer to args string.
-	struct task_struct *task = (void *)bpf_get_current_task();
-	char *args = (void *)BPF_CORE_READ(task, mm, arg_start);
-
-	// Read args to buffer.
-	char arg_buf[MAX_ARG_LEN]; // Allocate it out of ctx to pass CO-RE
-	struct get_real_comm_ctx ctx = {};
-
-	ctx.arg_buf = arg_buf;
-	ret = bpf_core_read_user_str(arg_buf, MAX_ARG_LEN, args);
-	if (unlikely(ret < 0)) {
-		bpf_printk(
-			"failed to read process name: bpf_core_read_user_str: %d",
-			ret);
-		return ret;
-	}
-
-	// Find range of command name.
-	ret = bpf_loop(MAX_ARG_LEN, get_real_comm_loop_cb, &ctx, 0);
-	if (unlikely(ret < 0))
-		return ret;
-
-	u8 offset = ctx.l;
-
-	for (u8 i = 0; i < TASK_COMM_LEN; i++) {
-		if (offset + i < MAX_ARG_LEN && arg_buf[offset + i] != '\0') {
-			pid_pname->pname[i] = arg_buf[offset + i];
-		} else {
-			pid_pname->pname[i] = '\0';
-			break;
-		}
-	}
-
-	// Pupulate tgid
-	ret = bpf_core_read(&pid_pname->pid, sizeof(pid_pname->pid),
-			    &task->tgid);
-	if (unlikely(ret < 0)) {
-		bpf_printk("failed to read pid: %d", ret);
-		return ret;
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	pid_pname->pid = pid_tgid >> 32;
+	if (bpf_get_current_comm(&pid_pname->pname, sizeof(pid_pname->pname))) {
+		pid_pname->pname[0] = '\0';
 	}
 	return 0;
 }
