--- a/noinst/libcrafter/libcrafter/configure.ac.anc	2018-07-20 16:45:10.760534965 +0200
+++ b/noinst/libcrafter/libcrafter/configure.ac	2018-07-20 16:46:03.339973975 +0200
@@ -29,21 +29,8 @@
      ;;
   *)
      AC_MSG_RESULT($withval)
-     if test -f $withval/pcap.h -a -f $withval/libpcap.a; then
-        owd=`pwd`
-        if cd $withval; then withval=`pwd`; cd $owd; fi
 	PCAPINC="-I$withval -I$withval/bpf"
 	PCAPLIB="-L$withval -lpcap"
-     elif test -f $withval/include/pcap.h -a \
-	       -f $withval/include/net/bpf.h -a \
-	       -f $withval/lib/libpcap.a; then
-	owd=`pwd`
-	if cd $withval; then withval=`pwd`; cd $owd; fi
-	PCAPINC="-I$withval/include"
-	PCAPLIB="-L$withval/lib -lpcap"
-     else
-        AC_ERROR(pcap.h, net/bpf.h, or libpcap.a not found in $withval)
-     fi
      ;;
   esac ],
 [ if test -f ${prefix}/include/pcap.h; then
