--- a/noinst/libcrafter/libcrafter/crafter/Protocols/DNSQuery.cpp	2018-07-20 17:28:21.252256760 +0200
+++ b/noinst/libcrafter/libcrafter/crafter/Protocols/DNSQuery.cpp	2018-07-20 17:28:35.092103432 +0200
@@ -85,7 +85,7 @@
 
 size_t DNS::DNSQuery::Compress() {
 	/* Put data into the buffer */
-	int nbytes = ns_name_compress(qname.c_str(), cqname, sizeof(cqname) , NULL,
+	int nbytes = dn_comp(qname.c_str(), cqname, sizeof(cqname) , NULL,
 			NULL);
 	if(nbytes == -1)
 		throw std::runtime_error("DNS::DNSQuery::Compress() : Error compressing the domain name provided");
--- a/noinst/libcrafter/libcrafter/crafter/Protocols/DNSAnswer.cpp	2018-07-20 17:37:46.614513337 +0200
+++ b/noinst/libcrafter/libcrafter/crafter/Protocols/DNSAnswer.cpp	2018-07-20 17:38:24.298131516 +0200
@@ -209,7 +209,7 @@
 
 size_t DNS::DNSAnswer::CompressName() {
 	/* Put data into the buffer */
-	int nbytes = ns_name_compress(qname.c_str(), cqname, sizeof(cqname), NULL,
+	int nbytes = dn_comp(qname.c_str(), cqname, sizeof(cqname), NULL,
 			NULL);
 	if(nbytes == -1)
 		throw std::runtime_error("DNSAnswer::CompressName() : Error compressing the domain name provided");
@@ -224,7 +224,7 @@
 size_t DNS::DNSAnswer::CompressRData() {
 	if (rdata.find_first_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIKKLMNOPQRSTUVWXYZ") != std::string::npos) {
 		/* Put data into the buffer */
-		int nbytes = ns_name_compress(rdata.c_str(), crdata, sizeof(crdata),
+		int nbytes = dn_comp(rdata.c_str(), crdata, sizeof(crdata),
 				NULL,  NULL);
 		if(nbytes == -1)
 			throw std::runtime_error("DNSAnswer::CompressRData() : Error compressing the domain name provided");
