--- a/src/tracebox/lua.cc	2018-07-20 18:03:00.439624791 +0200
+++ b/src/tracebox/lua.cc	2018-07-20 18:03:38.035263456 +0200
@@ -6,7 +6,12 @@
  */
 
 
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
 #include "lua/lua_packet.hpp"
 #include "config.h"
 
 extern lua_State* l_init();
--- a/src/tracebox/lua/lua_base.hpp	2018-08-06 09:43:49.589813406 +0200
+++ b/src/tracebox/lua/lua_base.hpp	2018-08-06 09:44:27.773454118 +0200
@@ -13,8 +13,12 @@
 #include <crafter.h>
 
 #define LUA_COMPAT_ALL
-#include <lua.hpp>
-#include <lua.h>
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
+
 
 #include "config.h"
 
