--- a/easyrsa3/easyrsa
+++ b/easyrsa3/easyrsa
@@ -1557,8 +1557,7 @@
 	prog_file="$0"
 	prog_file2="$(which -- "$prog_file" 2>/dev/null)" && prog_file="$prog_file2"
 	prog_file2="$(readlink -f "$prog_file" 2>/dev/null)" && prog_file="$prog_file2"
-	prog_dir="${prog_file%/*}"
-	prog_vars="${prog_dir}/vars"
+	prog_vars="/etc/easy-rsa/vars"
 	# set up PKI path
 	pki_vars="${EASYRSA_PKI:-$PWD/pki}/vars"
 
@@ -1588,9 +1587,9 @@
 	fi
 	
 	# Set defaults, preferring existing env-vars if present
-	set_var EASYRSA		"$prog_dir"
+	set_var EASYRSA		"/etc/easy-rsa"
 	set_var EASYRSA_OPENSSL	openssl
-	set_var EASYRSA_PKI	"$PWD/pki"
+	set_var EASYRSA_PKI	"/tmp/easyrsa3/pki"
 	set_var EASYRSA_DN	cn_only
 	set_var EASYRSA_REQ_COUNTRY	"US"
 	set_var EASYRSA_REQ_PROVINCE	"California"
